app/models/log_spreadsheet.rb

```diff
@@ -45,6 +44,14 @@ def self.count
     LogSpreadsheet.unscoped.count
   end
 
+  def self.remove_old_spreadsheets
+    keep = LogSpreadsheet.order(created_at: :desc).limit(SPREADSHEET_COUNT_LIMIT)
+    if keep.length > 0
+      limit = keep.last.created_at
+      LogSpreadsheet.where('created_at < ?', limit).destroy_all
```

Comment:
destroy_all will load in all of the objects in order to call their destroy callbacks. delete_all will just use SQL to delete all of the objects from the database without loading them in.  I don't see any callbacks in LogSpreadsheet or any dependencies that would get deleted, so I think delete_all is the best here.